edit/lint.js

```diff
@@ -0,0 +1,323 @@
+/* global CodeMirror CSSLint editors makeSectionVisible showHelp showCodeMirrorPopup */
+/* global stylelintDefaultConfig onDOMscripted injectCSS require */
+'use strict';
+
+function initLint() {
+  document.getElementById('lint-help').addEventListener('click', showLintHelp);
+  document.getElementById('lint').addEventListener('click', gotoLintIssue);
+  window.addEventListener('resize', resizeLintReport);
+  document.getElementById('stylelint-settings').addEventListener('click', openStylelintSettings);
+
+  // touch devices don't have onHover events so the element we'll be toggled via clicking (touching)
+  if ('ontouchstart' in document.body) {
+    document.querySelector('#lint h2').addEventListener('click', toggleLintReport);
+  }
+  BG.chromeLocal.getValue('editorStylelintRules').then(rules => setStylelintRules(rules));
+}
+
+function setStylelintRules(rules = []) {
+  if (Object.keys(rules).length === 0 && typeof stylelintDefaultConfig !== 'undefined') {
+    rules = deepCopy(stylelintDefaultConfig.rules);
+  }
+  BG.chromeLocal.setValue('editorStylelintRules', rules);
+  return rules;
+}
+
+function getLinterConfigForCodeMirror(name) {
+  return CodeMirror.lint && CodeMirror.lint[name] ? {
+    getAnnotations: CodeMirror.lint[name],
+    delay: prefs.get('editor.lintDelay')
+  } : false;
+}
+
+function updateLinter(name) {
+  function updateEditors() {
+    const options = getLinterConfigForCodeMirror(name);
+    CodeMirror.defaults.lint = options === 'null' ? false : options;
+    editors.forEach(cm => {
+      // set lint to "null" to disable
+      cm.setOption('lint', options);
+      cm.refresh(); // enabling/disabling linting changes the gutter width
+      updateLintReport(cm, 200);
+    });
+  }
+  if (prefs.get('editor.linter') !== name) {
+    prefs.set('editor.linter', name);
+  }
+  // load scripts
+  loadSelectedLinter(name).then(() => {
+    updateEditors();
+  });
+  $('#stylelint-settings').style.display = name === 'stylelint' ?
+    'inline-block' : 'none';
+}
+
+function updateLintReport(cm, delay) {
+  if (delay === 0) {
+    // immediately show pending csslint/stylelint messages in onbeforeunload and save
+    update(cm);
+    return;
+  }
+  if (delay > 0) {
+    setTimeout(cm => { cm.performLint(); update(cm); }, delay, cm);
```

Comment:
violates our `brace-style` eslint rule: multiple statements should be on multiple lines